google/cloud/firestore_v1/stream_generator.py

```diff
@@ -38,3 +65,35 @@ def throw(self, exp=None):
 
     def close(self):
         return self._generator.close()
+    
+    @property
+    def explain_options(self) -> ExplainOptions:
+        """Query profiling options for this stream request."""
+        return self._explain_options
+
+    @property
+    def explain_metrics(self) -> ExplainMetrics:
+        """
+        Get the metrics associated with the query execution.
+        Metrics are only available when explain_options is set on the query. If
+        ExplainOptions.analyze is False, only plan_summary is available. If it is
+        True, execution_stats is also available.
+        :rtype: :class:`~google.cloud.firestore_v1.query_profile.ExplainMetrics`
+        :returns: The metrics associated with the query execution.
+        :raises: :class:`~google.cloud.firestore_v1.query_profile.QueryExplainError`
+            if explain_metrics is not available on the query.
+        """
+        if self._explain_metrics is not None:
+            return self._explain_metrics
+        elif self._explain_options is None:
+            raise QueryExplainError("explain_options not set on query.")
+        elif self._explain_options.analyze is False:
+            # we need to run the query to get the explain_metrics
+            try:
+                next(self)
```

Comment:
Also, wouldn't the value from `next()` be lost here?

Reply:
I think when `analyze == True`,  the query isn't run so there's no query results returned anyway. `explain_metrics` is parsed when calling `next()`. Are there any other returned value that I'm not aware of?

Reply:
I actually copied from your PR. This is from your branch: https://github.com/googleapis/python-datastore/blob/e67ca728a982b15b6d2bc0f18fe9c27f07014471/google/cloud/datastore/query.py#L871-L872

Also updated the comments.